.github/api_best_practices.md

```diff
@@ -0,0 +1,231 @@
+and maintaining APIs. The guidelines helps identify potential pitfalls and design choices that could lead to issues to 
+help developers avoid common mistakes and build more robust APIs. When in doubt, It is encouraged to seek resolution
+in slack channel [#forum-acs-api-design](https://redhat-internal.slack.com/archives/C05MMG2PP8A).
+
+## Table of Contents
+
+- [Naming Guidelines](#naming-guidelines)
+  - [gRPC Service Name](#grpc-service-name)
+  - [gRPC Method Name](#grpc-method-name)
+  - [gRPC Message Name](#grpc-message-name)
+
+## Naming Guidelines
+
+### gRPC Service Name
+Service name **must** be unique and should use a noun that generally refers to a resource or product component 
+e.g. `DeploymentService`, `ReportService`, `ComplianceService`. Intuitive and well-known short forms or 
+abbreviations may be used in some cases (and could even be preferable) for succinctness 
+e.g. `ReportConfigService`, `RbacService`. 
+All gRPC methods grouped into a single service must generally pertain to the primary resource of the service.
+
+Note:
+- All service defined in versioned package [/proto/api/v2](https://github.com/stackrox/stackrox/blob/master/proto/v2)
+**must not** import from [/proto/storage](https://github.com/stackrox/stackrox/blob/master/proto/storage) and
+[/proto/internalapi](https://github.com/stackrox/stackrox/blob/master/proto/internalapi) packages.
+- All new gRPC methods defined in versioned package [/proto/api/v1](https://github.com/stackrox/stackrox/blob/master/proto/v1) 
+**must not** import from `/proto/storage` and `/proto/internalapi` packages.
+- All services of the type `APIServiceWithCustomRoutes` **must not** import from `/proto/storage` and `/proto/internalapi` packages.
+
+### gRPC Method Name
+Methods should be named such that they provide insights into the functionality. Typically, the method 
+name should follow the _VerbNoun_ convention. For example, `StartComplianceScan`, `RunComplianceScan`, 
+and `GetComplianceScan` are not the same. `StartComplianceScan` must return without waiting for the compliance 
+scan to complete. `RunComplianceScan` may or may not wait for a compliance scan; the method name is ambiguous 
+and should be avoided if it starts a process and returns without waiting. `GetComplianceScan` should not run 
+a compliance scan but only fetches a stored one.
+
+| Verb   | Noun           | Method name         |
+|--------|----------------|---------------------|
+| List   | Deployment     | `ListDeployments`   |
+| Get    | Deployment     | `GetDeployment`     |
+| Update | Deployment     | `UpdateDeployment`  |
+| Delete | Deployment     | `DeleteDeployment`  |
+| Notify | Violation      | `NotifyViolation`   |
+| Run    | ComplianceScan | `RunComplianceScan` |
+
+It is **recommended** that the verbs be imperative instead of inquisitive. Generally, the noun should be the resource type. 
```

Comment:
Do you intend to be consistent with the use of bolding to indicate the strength of the suggestion?  i.e. recommended, should, may, must.  I see various places where these words are not bolded.